lib/virt_autotest/utils.pm

```diff
@@ -857,4 +858,43 @@ sub recreate_guests {
     }
 }
 
+# For vms with imported disks, this function can be used to download the disk from ${OPENQA_URL}/assets/ to SUT.
+sub download_vm_import_disks {
+    my $download_dir = shift;
+
+    $download_dir //= "/var/lib/libvirt/images";
+    my $disks_to_download = get_var("VM_IMPORT_DISK_CONFIG", '');
+
+    if (!$disks_to_download) {
+        record_info "No import disk configured, skip download.";
+        return;
+    } else {
+        record_info "Going to download imported disk...";
+    }
+
+    assert_script_run("mkdir -p $download_dir");
+
+    my $BASE_URL = get_required_var("OPENQA_URL") . "/assets/";
+    foreach my $disk (split(/,/, $disks_to_download)) {
+        my $download_url = $BASE_URL . $disk;
+        die "URL is not accessible: $download_url." unless head($download_url);
+
+        $disk =~ /.*\/([^\/]+)\.([^\/\.]+)$/m;
```

Comment:
If $disk is just file name itself, the following is enough:
```
$disk =~ /^(.*)\.(.*)$/m;
```

Reply:
It looks like hdd/xxx.qcow2, or hdd/fixed/xx.xx.